### PR TITLE
fixes:fixed farcaster signin issue

### PR DESCRIPTION
**Pull request by sbvhev**

### CODE REVIEW

File: src/components/common/SignInButton.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import React from "react"
+import { SignInButton as SignInBtn, UseSignInArgs } from "@farcaster/auth-kit"
+
+type SignInButtonProps = UseSignInArgs & {
+  onSignOut?: () => void
+  debug?: boolean
+  hideSignOut?: boolean
+}
+
+const SignInButton: React.FC = ({ ...props }: SignInButtonProps) => {
```
Comment (tinaszheng) [inline src/components/common/SignInButton.tsx:11]: is this component still used?

File: src/components/common/Navbar.tsx [hunk 2]
```diff
@@ -46,6 +52,14 @@ const Navbar: React.FC = () => {
     }
   }, [error, toast])
 
+  const handleGetExpiryTime = useMemo(() => {
+    const currentDatetime = new Date()
```
Comment (tinaszheng) [inline src/components/common/Navbar.tsx:56]: `Date.now()` for getting the current time is slightly more performant!

File: src/components/common/Navbar.tsx [hunk 3]
```diff
@@ -1,41 +1,47 @@
...
+} from "@/components/ui/dropdown-menu"
 
 import { useToast } from "../ui/use-toast"
-
-import "@farcaster/auth-kit/styles.css"
+import { Button } from "../ui/button"
+import { Avatar, AvatarFallback, AvatarImage } from "../ui/avatar"
 
 const WelcomeDialog = dynamic(() => import("../WelcomeDialog"))
+const MemorizedSignInButton = React.memo(SignInButton)
```
Comment (tinaszheng) [inline src/components/common/Navbar.tsx:23]:
```suggestion
const MemoizedSignInButton = React.memo(SignInButton)
```

File: src/app/api/auth/[...nextauth]/route.ts [hunk 4]
```diff
@@ -41,7 +40,7 @@ export const handler = NextAuth({
         const verifyResponse = await appClient.verifySignInMessage({
           message: credentials?.message as string,
           signature: credentials?.signature as `0x${string}`,
-          domain: "example.com",
+          domain: "localhost:3000",
```
Comment (tinaszheng) [inline src/app/api/auth/[...nextauth]/route.ts:43]: this can probs be an env variable~

File: src/components/common/Navbar.tsx [hunk 5]
```diff
@@ -58,22 +72,71 @@ const Navbar: React.FC = () => {
...
+                onClick={() => signOut()}
+              >
+                Logout
+              </DropdownMenuItem>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        ) : (
+          <MemorizedSignInButton
+            nonce={getNonce}
+            expirationTime={handleGetExpiryTime}
```
Comment (tinaszheng) [inline src/components/common/Navbar.tsx:109]: does it make sense to have any sort of expiration checks here if we're going to read from the next-auth context for authenticated users?

**Review by bswags — APPROVED** [review]

**Pull request merged**